crates/astria-composer/src/searcher.rs

```diff
@@ -46,6 +58,10 @@ pub(super) struct Searcher {
     // The client for submitting wrapped and signed pending eth transactions to the astria
     // sequencer.
     sequencer_client: SequencerClient,
+    // Private key used to sign sequencer transactions
+    sequencer_key: SigningKey,
+    // Nonce of the sequencer account we sign with
+    sequencer_nonce: Arc<Mutex<Nonce>>,
```

Comment:
This would be better as an `Arc<AtomicU32>` (both to not depend on internal sequencer types, but also because the Mutex is heavy and IMO unnecessary).